src/assets/Generator.Shared/ArmOperationHelpers.cs

```diff
@@ -13,344 +13,344 @@
 namespace Azure.Core
 {
     /// <summary>
-    /// Helper methods for ARM long-running operations.
+    /// This implements the ARM scenarios for LROs. It is highly recommended to read the ARM spec prior to modifying this code:
+    /// https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/Addendum.md#asynchronous-operations
+    /// Other reference documents include:
+    /// https://github.com/Azure/autorest/blob/master/docs/extensions/readme.md#x-ms-long-running-operation
+    /// https://github.com/Azure/adx-documentation-pr/blob/master/sdks/LRO/LRO_AzureSDK.md
     /// </summary>
-    internal static class ArmOperationHelpers
+    /// <typeparam name="T">The final result of the LRO.</typeparam>
+    internal class ArmOperationHelpers<T>
```

Comment:
I'm a little confused by this. It is essentially what was `ArmOperation` but not a type of `Operation<T>`. Is there a reason for that? You're already using this as a composition-based operation field inside an operation.

Reply:
There is just no reason for it to implement Operation<T> and I don't want anyone to accidentally use it directly.